### ISSUE TITLE
max-len fails to identify lines in violation that contain long strings

### ISSUE THREAD

**Issue by robertmain** [eslint/eslint#11682]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 10.9.0
* **npm Version:** 6.8.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "parser": "babel-eslint",
  "extends": [],
  "env": {
    "browser": true,
    "node": true,
    "mocha": true
  },
  "plugins": [
    "react",
    "mocha"
  ],
  "rules": {
    "max-len": [
      "error",
      {
        "code": 80,
        "ignoreStrings": true,
        "tabWidth": 2
      }
    ],
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Setting `rules.max-len.ignoreStrings` to `false` causes the second example to be correctly reported by eslint. Setting it to `true` causes the second example to be ignored.

```js
          const foo = bar.baz.fubar.fubar.fubar.fubar.fubar.fubar.fubar.fubar.fubar; // Reported
          const foo = bar.baz.fubar.fubar.fubar.fubar.fubar.fubar.fubar.fubar('string param'); // Ignored
```

```bash
npx eslint src/file.js
```

**What did you expect to happen?**

The second example should also be reported as being in violation of the rule.

**What actually happened? Please include the actual, raw output from ESLint.**
```
/home/robert/Code/project/src/file.js
   1:1  error  Line 1 exceeds the maximum line length of 80   max-len
```

**Are you willing to submit a pull request to fix this bug?**

**Comment by kaicataldo** [comment]
From the [max-len docs](https://eslint.org/docs/rules/max-len#options):
>`"ignoreStrings": true ignores lines that contain a double-quoted or single-quoted string`

It looks like the `ignoreStrings` option is working as intended to me, as there is a string in the second example.

**Comment by robertmain** [comment]
That's kind of my point though. Shouldn't that option only ignore lines with only strings - rather than lines that contain a string somewhere? Otherwise, large chunks of your code could be in violation and you wouldn't know if they contain a string.

**Comment by kaicataldo** [comment]
@robertmain Sorry for misunderstanding. I thought you were reporting a bug rather than proposing a change in behavior. So I understand, this is the behavior you're advocating for is as follows (please correct me if I'm wrong):

```js
/*eslint max-len: ["error", { "code": 80, ignoreStrings: true }]*/

const foo = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris sollicitudin,' + // This should warn
  ' nisi non varius lobortis, lacus lacus consequat libero, a dignissim leo nisi non lectus. Sed vehicula sodales dolor id dictum'; // This should not warn
```

```js
/*eslint max-len: ["error", { "code": 80, ignoreStrings: true }]*/

const foo = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris sollicitudin,' // This should warn
  + ' nisi non varius lobortis, lacus lacus consequat libero, a dignissim leo nisi non lectus. Sed vehicula sodales dolor id dictum'; // This should now also warn
```

```js
/*eslint max-len: ["error", { "code": 80, ignoreStrings: true }]*/

const foo = 'Lorem ipsum' + // This should not warn
  'dolor sit amet, consectetur adipiscing elit. Mauris sollicitudin, nisi non varius lobortis, lacus lacus consequat libero, a dignissim leo nisi non lectus. Sed vehicula sodales dolor id dictum'; // This should also not warn
```

For context, here was the original discussion around the inclusion of this option: https://github.com/eslint/eslint/issues/5805. In practice, the behavior outlined above doesn't seem to me to be that different from the pre-existing behavior of the rule without using the option at all, since you're still forced to use string concatenation and, in the event of the string changing, might have to move things between lines.

Regardless, if you'd like to propose a behavioral change, would you mind following the directions for proposing rule changes as described [here](https://eslint.org/docs/developer-guide/contributing/rule-changes)? Thanks!

**Comment by robertmain** [comment]
I thought I was reporting a bug too. I guess not. Thanks :)

**Comment by kaicataldo** [comment]
@robertmain If you'd like to not use the rule with that behavior, you can always remove `ignoreStrings: true` from your `max-len` config.
